packages/engine-test-utils/src/__tests__/dendron-cli/commands/doctor.spec.ts

```diff
@@ -122,6 +123,60 @@ const runDoctor = (opts: Omit<DoctorCLICommandOpts, "server">) => {
     server: {} as any,
   });
 };
+describe("FIX_FRONTMATTER", () => {
+  const action = DoctorActionsEnum.FIX_FRONTMATTER;
+  describe("GIVEN note with missing frontmatter", () => {
+    test("THEN generate frontmatter ", async () => {
+      await runEngineTestV5(
+        async ({ engine, wsRoot, vaults }) => {
+          await runDoctor({
+            wsRoot,
+            engine,
+            action,
+          });
+          const fname = "test";
+          const vault = vaults[0];
+          const fpath = path.join(vault2Path({ vault, wsRoot }), `${fname}.md`);
+          await checkFile({ fpath }, "---", "id:");
+        },
+        {
+          expect,
+          preSetupHook: async (opts) => {
+            TestDoctorUtils.createNotesWithNoFrontmatter(opts);
+          },
+        }
+      );
+    });
+  });
+
+  describe("GIVEN note with invalid frontmatter id", () => {
+    test("THEN replace frontmatter id", async () => {
+      await runEngineTestV5(
+        async ({ engine, wsRoot, vaults }) => {
+          await runDoctor({
+            wsRoot,
+            engine,
+            action,
+          });
+          const fname = "test";
+          const vault = vaults[0];
+          const note = NoteUtils.getNoteByFnameFromEngine({
+            fname,
+            vault,
+            engine,
+          })!;
+          expect(note.id === "-bad-id").toBeFalsy();
+        },
+        {
+          expect,
+          preSetupHook: async (opts) => {
+            TestDoctorUtils.createNotesWithBadIds(opts);
+          },
+        }
+      );
+    });
+  });
+});
```

Comment:
Could you also add a test where you check it with different scopes? E.g. if you run the doctor action with a scope of 1 note, it should only replace the id in that note and not other notes.